demos/python_demos/formula_recognition_demo/formula_recognition_demo.py

```diff
@@ -21,63 +21,170 @@
+        frame = cv.putText(frame, "Predicted:", comment_coords,
+                           fontFace=cv.FONT_HERSHEY_SIMPLEX, fontScale=0.7, color=COLOR_WHITE, thickness=2, lineType=cv.LINE_AA)
+        return frame
+
+    def put_crop(self, frame, crop):
+        height = self.end_point[1] - self.start_point[1]
+        width = self.end_point[0] - self.start_point[0]
+        crop = cv.resize(crop, (width, height))
+        frame[0:height, self.start_point[0]:self.end_point[0], :] = crop
+        comment_coords = (0, 20)
+        frame = cv.putText(frame, "Model input:", comment_coords, fontFace=cv.FONT_HERSHEY_SIMPLEX,
+                           fontScale=0.7, color=COLOR_WHITE, thickness=2, lineType=cv.LINE_AA)
+        return frame
+
+    def _put_formula_img(self, frame, formula):
+        if self._renderer is None or formula == '':
+            return frame
+        formula_img = self._render_formula_async(formula)
+        if formula_img is None:
+            return frame
+        y_start = self.end_point[1] - self.start_point[1] + self._latex_h * 2
+        formula_img = self._resize_if_need(formula_img)
+        frame[y_start:y_start + formula_img.shape[0],
+              self.start_point[0]:self.start_point[0] + formula_img.shape[1],
+              :] = formula_img
+        comment_coords = (0, y_start + (formula_img.shape[0] + self._latex_h) // 2)
+        frame = cv.putText(frame, "Rendered:", comment_coords,
+                           fontFace=cv.FONT_HERSHEY_SIMPLEX, fontScale=0.7, color=COLOR_WHITE, thickness=2, lineType=cv.LINE_AA)
+        return frame
+
+    def _resize_if_need(self, formula_img):
+        if (self.end_point[0] - self.start_point[0]) < formula_img.shape[1]:
+            scale_factor = (self.end_point[0] - self.start_point[0]) / formula_img.shape[1]
+            formula_img = cv.resize(formula_img, fx=scale_factor, fy=scale_factor, dsize=None)
+        return formula_img
+
+    def _render_formula_async(self, formula):
+        if formula == self._prev_rendered_formula:
+            return self._prev_formula_img
+        result = self._renderer.thread_render(formula)
+        if result is None:
+            return None
+        formula_img, res_formula = result
+        if res_formula != formula:
+            return None
+        self._prev_rendered_formula = formula
+        self._prev_formula_img = formula_img
+        return formula_img
+
+    def draw(self, frame, phrase):
+        frame = self._put_text(frame, phrase)
+        frame = self._put_formula_img(frame, phrase)
+        frame = self._draw_rectangle(frame)
+        return frame
+
+
+def create_capture(input_source, demo_resolution):
+    if input_source.isdigit():
+        try:
+            input_source = int(input_source)
+        except ValueError:
+            raise ValueError("\n \tCannot recognize input as .mp4 video or device-id as int"
+                             "\n \tPlease, check the -i/--input arg and try again")
```

Comment:
```suggestion
    try:
        input_source = int(input_source)
    except ValueError:
        pass
```

(I don't think that your message can ever be displayed, other than if `input_source` includes non-Arabic digits, in which case it should be treated as a path anyway.)